strictdoc/core/file_traceability_index.py

```diff
@@ -990,3 +939,147 @@ def compare_sdocnode_by_uid(node_: SDocNode) -> str:
                 return assert_cast(node_.reserved_uid, str)
 
             path_requirements_.sort(key=compare_sdocnode_by_uid)
+
+    def connect_source_node_function(
+        self,
+        source_node: SourceNode,
+        source_sdoc_node_uid: str,
+        traceability_info: SourceFileTraceabilityInfo,
+    ) -> None:
+        source_node_function = source_node.function
+        assert source_node_function is not None
+
+        function_marker = self.forward_function_marker_from_function(
+            function=source_node_function,
+            marker_type=RangeMarkerType.FUNCTION,
+            reqs=[Req(None, source_sdoc_node_uid)],
+            role=None,
+            description=f"function {source_node_function.display_name}()",
+        )
+
+        traceability_info.ng_map_reqs_to_markers.setdefault(
+            source_sdoc_node_uid, []
+        ).append(function_marker)
+        function_marker_copy = function_marker.create_end_marker()
+        traceability_info.markers.append(function_marker)
+        traceability_info.markers.append(function_marker_copy)
+
+    @staticmethod
+    def create_sdoc_node_from_source_node(
+        source_node: SourceNode,
+        uid: str,
+        parent_document: SDocDocumentIF,
+        relevant_source_node_entry: dict[str, str],
+    ) -> SDocNode:
+        source_sdoc_node = SDocNode(
+            parent=parent_document,
+            node_type=relevant_source_node_entry["node_type"],
+            fields=[],
+            relations=[],
+            # It is important that this autogenerated node is marked as such.
+            autogen=True,
+        )
+        source_sdoc_node.ng_document_reference = DocumentReference()
+        source_sdoc_node.ng_document_reference.set_document(parent_document)
+        source_sdoc_node.ng_including_document_reference = DocumentReference()
+        source_sdoc_node.set_field_value(
+            field_name="UID",
+            form_field_index=0,
+            value=uid,
+        )
+        source_sdoc_node.set_field_value(
+            field_name="TITLE",
+            form_field_index=0,
+            value=source_node.entity_name,
+        )
+        for node_name_, node_value_ in source_node.fields.items():
+            source_sdoc_node.set_field_value(
+                field_name=node_name_,
+                form_field_index=0,
+                value=node_value_,
+            )
+        return source_sdoc_node
+
+    @staticmethod
+    def merge_sdoc_node_with_source_node(
+        sdoc_node: SDocNode,
+        source_node: SourceNode,
+        parent_document: SDocDocumentIF,
+        source_node_config_entry: dict[str, str],
+    ) -> None:
+        # we can only merge if grammar element definitions are compatible
```

Comment:
Please make all comments start with a capital letter and end with a dot. 

(There is a linter tool in the repo to check this but due to some false positives I haven't activated it in the CI yet.)